### ISSUE TITLE
Setup COMET code for hyperparam search

### ISSUE THREAD

**Issue by radka-j** [alan-turing-institute/ARC-MTQE#44]
This should allow varying:
- [x] learning rate
- [x] dropout
- [x] loss function (CrossEntropy or BinaryCrossEntropyWithLogits - both have final activation included in the calculation) &
    - [x] output dimension (1 or 2)
- [x] batch size
- [x] freezing/unfreezing of the encoder
- [x] allow random initialisation of head (vs just loading pre-trained weights)
- [x] oversampling of data
- [x] weighted loss function

Some of the above parameters (loss function, batch size) will be fixed before doing more comprehensive search over the other hyperparameters.
